stream_alert/athena_partition_refresh/main.py

```diff
@@ -204,37 +233,178 @@ def check_table_exists(self, table_name):
+
             query_success, query_resp = self.run_athena_query(
                 query='MSCK REPAIR TABLE {};'.format(athena_table),
                 database=self.DATABASE_STREAMALERT
             )
+
             if query_success:
                 LOGGER.info('Query results:')
                 for row in query_resp['ResultSet']['Rows']:
                     LOGGER.info(row['Data'])
             else:
                 LOGGER.error('Partition refresh of the Athena table '
                              '%s has failed.', athena_table)
+                return False
+
+        return True
 
     @staticmethod
-    def firehose_partition_refresh(_):
-        """Execute a Firehose specific partition update"""
-        LOGGER.error('Firehose partition refresh is not yet supported, exiting!')
+    def add_hive_partition(_):
+        """Execute a Hive Add Partition command on a given Athena table"""
+        LOGGER.error('Add Hive Parition is not yet supported, exiting!')
         raise NotImplementedError
 
 
-def handler(event, _):
+class StreamAlertSQSClient(object):
+    """A StreamAlert SQS Client for polling and deleting S3 event notifications
+
+    Attributes:
+        config: The loaded StreamAlert configuration
+        sqs_client: The Boto3 SQS client
+        athena_sqs_url: The URL to the Athena SQS Queue
+        received_messages: A list of receieved SQS messages
+        processed_messages: A list of processed SQS messages
+    """
+    QUEUENAME = 'streamalert_athena_data_bucket_notifications'
+
+    def __init__(self, **kwargs):
+        self.config = kwargs['config']
+        self.received_messages = []
+        self.processed_messages = []
+
+        self.setup()
+
+    def setup(self):
+        """Get the SQS URL for Athena bucket s3 notifications"""
+        region = self.config['global']['account']['region']
+        self.sqs_client = boto3.client('sqs', region_name=region)
+
+        self.athena_sqs_url = self.sqs_client.list_queues(
+            QueueNamePrefix=self.QUEUENAME
+        )['QueueUrls'][0]
+
+    def get_messages(self):
+        """Poll the queue for messages"""
+        @backoff.on_predicate(backoff.fibo,
+                              # Backoff up to 5 times
+                              max_tries=5,
+                              # Don't backoff for longer than 5 seconds
+                              # This constrains the total max backoff to 25 seconds
+                              max_value=5,
+                              jitter=backoff.full_jitter,
+                              on_backoff=_backoff_handler,
+                              on_success=_success_handler)
+        def _receive_messages():
+            polled_messages = self.sqs_client.receive_message(
+                QueueUrl=self.athena_sqs_url,
+                MaxNumberOfMessages=10
+            )
+
+            if 'Messages' in polled_messages:
+                self.received_messages.extend(polled_messages['Messages'])
```

Comment:
suggest getting rid of the `else` by doing `if not 'Messages' in polled_messages: return False`